### ISSUE TITLE
[BUG][TYPESCRIPT] Property names should not be sanitized, but quoted instead

### ISSUE THREAD

**Issue by TimoGlastra** [OpenAPITools/openapi-generator#4748]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

As a developer I expect the `modelPropertyNaming='original'` to keep property names equal to the name provided in the openapi spec file. Not all variable names are accepted in TypeScript (and JavaScript) so for variable names it is expected to sanitize the names to generate valid code. For property names this is not the case, but the property names are still [sanitized and altered](https://github.com/OpenAPITools/openapi-generator/blob/b1984f3c49b6807d4b17c505574d579d9e5969d9/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java#L220). This should only happen when not doing this would produce invalid code. This is not the case, property names can be quoted to allow all possible names.

At the moment this generates clients that are not interoperable with the server

Actual:

```typescript
/**
 * 
 * @export
 * @interface Example
 */
export interface Example {
    /**
     * 
     * @type {string}
     * @memberof Example
     */
    type?: string;
    /**
     * 
     * @type {string}
     * @memberof Example
     */
    mime_type?: string;
}
```

Expected:

```typescript
/**
 * 
 * @export
 * @interface Example
 */
export interface Example {
    /**
     * 
     * @type {string}
     * @memberof Example
     */
    "@type"?: string;
    /**
     * 
     * @type {string}
     * @memberof Example
     */
    "mime-type"?: string;
}
```

##### openapi-generator version

`4.1.2`

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0
info:
  title: modelNamingExample
  version: v1
paths:
  /example:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/Example"
      responses:
        "200":
          description: ""
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Example"

components:
  schemas:
    Example:
      description: ""
      type: object
      properties:
        "@type":
          type: string
        "mime-type":
          type: string
```

##### Command line used for generation

Note: This is not only limited to the `typescript-axios` generator but all `typescript-*` generators. The [AbstractTypeScriptClientCodegen.java](https://github.com/swagger-api/swagger-codegen/blob/a408204a030ecb73e6562657660be9afe595765c/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java) file, which is used in all / most TypeScript generators, causes the problem.

```bash
openapi-generator generate \
    -i example.yaml \
    -g typescript-axios \
    -o generated-sources/openapi \
    --additional-properties=supportsES6=true,modelPropertyNaming='original'
```

##### Steps to reproduce

1. Generate API with above command and yaml file.
2. Check `Example` interface in generated `api.ts` file.

##### Related issues/PRs

I know there are some similar issues, but those are more on a specific case, like an `@` being removed or a `-` being replaced by an `_`. This issue is more focussed on the sanitizing of property names in general which shouldn't happen unless needed.

https://github.com/OpenAPITools/openapi-generator/issues/4065

##### Suggest a fix

I would suggest wrapping all property names (so not variable names) in quotes. You could argue that to keep it clean you should only wrap property names that require quotes to be valid should have quotes, but it would add complexity that is not really needed.

**Comment by ffMathy** [comment]
I agree with this. Everything should always be wrapped in quotes. There is no downside to doing so.

**Comment by vitorcamachoo** [comment]
Any update related with this issue ?

**Comment by smorandi** [comment]
yes please, this would really be helpful

**Comment by Morriz** [comment]
Why is this so problematic yet not much activity here?

**Comment by ffMathy** [comment]
Because everyone is switching to Autorest 😅 I did when I encountered this. This repo has severe quality issues.

**Comment by TimoGlastra** [comment]
Same here

**Comment by smorandi** [comment]
hmm, looks very interesting indeed

**Comment by carloslabrador** [comment]
Same here

**Comment by HomiG** [comment]
Seems like an interesting and yet simple issue to be open.
